core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala

```diff
@@ -78,6 +80,7 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
     }
     // Note: use getSizeAsKb (not bytes) to maintain compatibility if no units are provided
     blockSize = conf.getSizeAsKb("spark.broadcast.blockSize", "4m").toInt * 1024
+    chunkSize = SparkEnv.get.memoryManager.pageSizeBytes.toInt
     checksumEnabled = conf.getBoolean("spark.broadcast.checksum", true)
```

Comment:
@JoshRosen We use `SparkEnv.get.memoryManager.pageSizeBytes` as chunk size. As `SparkEnv.get.memoryManager.pageSizeBytes` returns `Long`, there is still underlying integer overflow issue, isn't it? Besides, users will never know the  low level details and the effect to chunk size when modify `pageSizeBytes`.